include/solvers/mpm_base.tcc

```diff
@@ -762,16 +762,44 @@ bool mpm::MPMBase<Tdim>::initialise_math_functions(const Json& math_functions) {
       const std::string function_type =
           function_props["type"].template get<std::string>();
 
+      // Initiate another function_prop to be passed
+      auto function_props_update = function_props;
+
+      // Create a file reader
+      const std::string io_type =
+          io_->json_object("mesh")["io_type"].template get<std::string>();
+      auto reader = Factory<mpm::IOMesh<Tdim>>::instance()->create(io_type);
+
+      // Math function is specified in a file, replace function_props_update
```

Comment:
Read files in `io` class not in `mpm_base`. Create a separate class in `io` or use the `read_mesh_ascii` to read these files. I'd prefer for now if you use `read_mesh_ascii` to read the input file.

Reply:
That's what I have tried to use in the beginning, and you requested CSV. Can we agree on which one to use now before I make more changes?

Reply:
I'm not asking you to revert back, this is an `io` operation and reading the CSV should be in the `io` class as an additional function and not in the `mpm_base`

Reply:
I see, so still CSV but within `read_mesh_ascii`. But doesn't it refer to `ascii` not `csv`?